### PR TITLE
Controller refactor

### PR DESCRIPTION
**Pull request by zacharyzjamison**
Couple of things were moved.

- router functionality in its own directory
- group user api in router in such a way you can add others
- user controller in its own directory
- database initialization in its own directory
- main file reduced to only calling database initialization and create server router

### CODE REVIEW

File: server/router/router.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+)
+
+func InitRouter() {
+
+	r := gin.Default()
+
+	router := r.Group("/v1")
+	{
+		// users
+		user_router := router.Group("/user")
```
Comment (Julius-R) [inline server/router/router.go]: In Go, camel case is the preferred convention.
https://go.dev/doc/effective_go#mixed-caps

File: server/controllers/UserController.go
Comment (Julius-R) [inline server/controllers/UserController.go:1]: I think the errors that are logged as fatal should be sent as http errors. That way the server hoster doesn't have to restart if a user makes a mistake sending a request.

File: server/models/user.go [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package models
 
 type User struct {
-	Id       int    `json:"id"`
+	Id       string `json:"id"`
 	Email    string `json:"email"`
 	Username string `json:"username"`
+	Password string `json:"-"`
```
Comment (Julius-R) [inline server/models/user.go:7]:
```suggestion
	Password string `json:"password,omitempty"`
```

Comment (Julius-R) [reply server/models/user.go:7]: This way, if the field doesn't exist, it's ignored

Comment (zacharyzjamison) [reply server/models/user.go:7]: @Julius-R will this only hide it if its empty? I want to hide the password in the response for all cases

Comment (Julius-R) [reply server/models/user.go:7]: oooh, I gotcha. No, it only hides empty,

File: server/router/router.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		{
+			user_router.POST("/", controllers.InsertUser)
+			user_router.GET("/", controllers.GetUsers)
+			user_router.GET("/:id", controllers.GetUser)
+			user_router.PATCH("/", controllers.UpdateUser)
+			user_router.DELETE("/", controllers.DeleteUser)
```
Comment (Julius-R) [inline server/router/router.go]: Could we also patch and delete by id, or is that not recommended?

Comment (zacharyzjamison) [reply server/router/router.go]: @Julius-R  I actually forgot about that. It should be based on the Id

File: server/router/router.go (hunk 1 above)
Comment (zacharyzjamison) [reply server/router/router.go]: @Julius-R updated variable to camelCase

File: server/controllers/UserController.go
Comment (zacharyzjamison) [reply server/controllers/UserController.go:1]: @Julius-R updated errors to send HTTP response.

File: server/router/router.go (hunk 3 above)
Comment (zacharyzjamison) [reply server/router/router.go]: @Julius-R PATCH and DELETE are now update to query by id

**Review by Julius-R — APPROVED** [review]

**Pull request merged**